restli-tools/src/main/java/com/linkedin/restli/tools/data/ExtensionSchemaValidationCmdLineApp.java

```diff
@@ -132,7 +139,29 @@ private static void parseAndValidateExtensionSchemas(String resolverPath, File i
           .stream()
           .filter(f -> !((RecordDataSchema) topLevelDataSchema).isFieldFromIncludes(f))
           .collect(Collectors.toList());
-      // TODO : Call annotation process to validate each field's annotation, need to create schemaVisitor and handler for extension annotation.
+
+      for (RecordDataSchema.Field field : extensionSchemaFields) {
+        Map<String, Object> properties = field.getProperties();
+
+        if (properties.isEmpty() || properties.keySet().size() != 1 || !properties.containsKey(
+            _extension_annotation_namespace)) {
+          _logger.error("The field [{}] of extension schema must and only be annotated with 'extension'", field.getName());
+          System.exit(1);
+        }
+      }
+
+      // Using annotation framework to check the annotations of fields in extension schema.
+      List<SchemaAnnotationHandler> handlers = schemaAnnotationHandlerUtil.getSchemaAnnotationHandlers(handlerJarPaths, handlerClassNames);
+
+      SchemaAnnotationProcessor.SchemaAnnotationProcessResult result =
+          SchemaAnnotationProcessor.process(handlers, topLevelDataSchema, new SchemaAnnotationProcessor.AnnotationProcessOption());
+      if (result.hasError())
+      {
+        _logger.error("Annotation processing for schema [{}] failed, detailed error: \n",
```

Comment:
In your case, should we reword to "annotation validation" instead of "annotation processing"?

Reply:
Updated log.